boot/jaxrs/src/main/java/org/commonjava/indy/boot/jaxrs/IndyDeployer.java

```diff
@@ -0,0 +1,136 @@
+package org.commonjava.indy.boot.jaxrs;
+
+import io.undertow.Undertow;
+import io.undertow.predicate.Predicate;
+import io.undertow.predicate.Predicates;
+import io.undertow.server.handlers.encoding.ContentEncodingRepository;
+import io.undertow.server.handlers.encoding.DeflateEncodingProvider;
+import io.undertow.server.handlers.encoding.EncodingHandler;
+import io.undertow.server.handlers.encoding.GzipEncodingProvider;
+import io.undertow.servlet.Servlets;
+import io.undertow.servlet.api.DeploymentInfo;
+import io.undertow.servlet.api.DeploymentManager;
+import org.commonjava.indy.bind.jaxrs.IndyDeployment;
+import org.commonjava.propulsor.boot.BootOptions;
+import org.commonjava.propulsor.boot.PortFinder;
+import org.commonjava.propulsor.deploy.DeployException;
+import org.commonjava.propulsor.deploy.Deployer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+
+import java.util.Collection;
+
+@ApplicationScoped
+public class IndyDeployer
+                implements Deployer
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    private Undertow server;
+
+    public IndyDeployer()
+    {
+    }
+
+    @Inject
+    private IndyDeployment indyDeployment;
+
+    @Override
+    public void stop()
+    {
+        if ( server != null )
+        {
+            server.stop();
+        }
+    }
+
+    @Override
+    public void deploy( BootOptions bootOptions ) throws DeployException
+    {
+        final DeploymentInfo di = indyDeployment.getDeployment( bootOptions.getContextPath() ).setContextPath( "/" );
+
+        final DeploymentManager dm = Servlets.defaultContainer().addDeployment( di );
+
+        Collection<String> list = Servlets.defaultContainer().listDeployments();
+        logger.info( "List deployments: {}", list );
+
+        dm.deploy();
+
+        try
+        {
+            Integer port = bootOptions.getPort();
+            if ( port < 1 )
+            {
+                logger.info( "Looking for open Undertow port..." );
+
+                final ThreadLocal<ServletException> errorHolder = new ThreadLocal<>();
```

Comment:
Can we use AtomicReference here instead of ThreadLocal?

Reply:
+1